libass/ass_fontselect.c

```diff
@@ -668,8 +668,11 @@ static bool matches_full_or_postscript_name(ASS_FontInfo *f,
 
     if (check_postscript(f))
         return matches_postscript_name;
-    else
-        return matches_fullname;
+
+    if (matches_postscript_name)
+        return matches_postscript_name;
```

Comment:
In particular, this doesn’t seem right. Here, `matches_postscript_name` is very intentionally disregarded if `check_postscript()` is not true; otherwise, this could be just `return matches_postscript_name || matches_fullname`.

This needs to be tested against VSFilter (I’ll do it if nobody else does before me) using the fonts from #386 and #578, because there’s a small chance that Windows handles variable fonts differently, but for normal fonts, the very point of this function is to match
* _either_ the PostScript name (and ignore the full name) for CFF-outline fonts (which use Adobe’s font driver on Windows)
* _or_ the full name (and ignore the PostScript name) for TrueType fonts (which use Microsoft’s font driver on Windows).